.github/workflows/snapshot.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish Snapshot to Docker Hub
+
+on:
+  push:
+    branches:
+      - master
+
+concurrency:
+  group: ${{ github.ref }}-snapshot
+  cancel-in-progress: true
```

Comment:
is this really necessary? how could this be triggered multiple times for the same ref?